src/app/core/models/repo.model.ts

```diff
@@ -13,7 +13,8 @@ export class Repo {
   }
 
   /** Creates a new Repo from one repository url. */
-  public static of(repoUrl: string) {
+  public static of(fullRepoUrl: string) {
```

Comment:
This param name gives me the feeling that only the full url (`https://github.com/org/repo`) is passed in. Maybe name it something clearer?

```suggestion
  public static of(repoUrlInput: string) {
```